huaweicloud/config/auth.go

```diff
@@ -0,0 +1,374 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/chnsz/golangsdk"
+	huaweisdk "github.com/chnsz/golangsdk/openstack"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
+	"github.com/jmespath/go-jmespath"
+	"github.com/mitchellh/go-homedir"
+)
+
+const (
+	securityKeyURL     string = "http://169.254.169.254/openstack/latest/securitykey"
+	keyExpiresDuration int64  = 600
+)
+
+// CLI Shared Config
+type SharedConfig struct {
+	Current  string    `json:"current"`
+	Profiles []Profile `json:"profiles"`
+}
+
+type Profile struct {
+	Name             string `json:"name"`
+	Mode             string `json:"mode"`
+	AccessKeyId      string `json:"accessKeyId"`
+	SecretAccessKey  string `json:"secretAccessKey"`
+	SecurityToken    string `json:"securityToken"`
+	Region           string `json:"region"`
+	ProjectId        string `json:"projectId"`
+	DomainId         string `json:"domainId"`
+	AgencyDomainId   string `json:"agencyDomainId"`
+	AgencyDomainName string `json:"agencyDomainName"`
+	AgencyName       string `json:"agencyName"`
+}
+
+func buildClient(c *Config) error {
+	if c.Token != "" {
+		return buildClientByToken(c)
+	} else if c.AccessKey != "" && c.SecretKey != "" {
+		return buildClientByAKSK(c)
+	} else if c.Password != "" && (c.Username != "" || c.UserID != "") {
+		return buildClientByPassword(c)
+	} else if c.SharedConfigFile != "" {
+		return buildClientByConfig(c)
+	}
+
+	return buildClientByMeta(c)
+}
+
+func genClient(c *Config, ao golangsdk.AuthOptionsProvider) (*golangsdk.ProviderClient, error) {
+	client, err := huaweisdk.NewClient(ao.GetIdentityEndpoint())
+	if err != nil {
+		return nil, err
+	}
+
+	// Set UserAgent
+	client.UserAgent.Prepend("terraform-provider-iac")
+
+	config, err := generateTLSConfig(c)
```

Comment:
suggest to move **generateTLSConfig** into auth.go

Reply:
done